corporate/lib/stripe.py

```diff
@@ -1439,6 +1542,18 @@ def do_update_plan(self, update_plan_request: UpdatePlanRequest) -> None:
             elif status == CustomerPlan.FREE_TRIAL:
                 assert plan.status == CustomerPlan.DOWNGRADE_AT_END_OF_FREE_TRIAL
                 do_change_plan_status(plan, status)
+            elif status == CustomerPlan.SWITCH_NOW_FREE_TRIAL_FROM_ANNUAL_TO_MONTHLY_CYCLE:
+                assert plan.billing_schedule == CustomerPlan.ANNUAL
+                assert plan.status == CustomerPlan.FREE_TRIAL
+                do_change_plan_status(
+                    plan, CustomerPlan.SWITCH_NOW_FREE_TRIAL_FROM_ANNUAL_TO_MONTHLY_CYCLE
+                )
+            elif status == CustomerPlan.SWITCH_NOW_FREE_TRIAL_FROM_MONTHLY_TO_ANNUAL_CYCLE:
+                assert plan.billing_schedule == CustomerPlan.MONTHLY
+                assert plan.status == CustomerPlan.FREE_TRIAL
+                do_change_plan_status(
+                    plan, CustomerPlan.SWITCH_NOW_FREE_TRIAL_FROM_MONTHLY_TO_ANNUAL_CYCLE
+                )
```

Comment:
@laurynmm can you review the main commit adding support for changing the billing cycle?

Reply:
@laurynmm the conflicts are from the changes in 39bfab0ec4bfd5aa2de0defac669f792518f2761. So, Let me know if you think we should do something different here.

Reply:
See review comments below. And let me know if you'd like me to push my local rebase for the merge conflicts between my changes and yours.